### DISCUSSION TITLE
Long times to redirect to BTCPay server payment page

### DISCUSSION THREAD

**Discussion by BakungaBronson** [btcpayserver/prestashop-plugin#143] [Technical Support]
Greetings @BitcoinMitchell and the team, I'm not sure if this is an issue or a misconfiguration on my end so I decided to put it here to get some clarity.
I've been using this prestashop-plugin for a few months now, first with the older version that did not officially support Prestashop 8.0, but I have now gone ahead and updated; however, there seems to still be a lag of roughly 16-21 seconds between the submission of the cart by the user and the time the redirection to the payment link via the BTCPay Server. Have you observed this and can you offer any guidance?

More information:
- I have checked, and all the required PHP modules are working.
- The Prestashop instance and the BTCPay Server are on two different instances.

Technical information:
PrestaShop 8.0.1
BTCPay Server v1.12.4
Symfony 4.4.49
PHP version8.1.27 
PHP ExtensionsxdebugAPCuOPcache
PHP SAPIfpm-fcgi

Image:
<img width="1119" alt="image" src="https://github.com/user-attachments/assets/736b457c-52d5-4ecf-8b5d-3712e3720f02">

**Comment by BitcoinMitchell** [comment]
I have the same issue when, for example, I check orders for the first time in a while. Once it gets going, it keeps being pretty quick, but the first request always take a while. 

Sadly I have no real answer for you, as far as I can tell, this delay is caused by BTCPay Server itself. Will look into it some more when I have time though! Will add some logging to see what is taking so long to load.

**Reply by BakungaBronson** [reply]
I see that makes sense.  I'll do some digging as well. Weirdly, the same behaviour doesn't exist when using the API directly.
